addons/sale_stock/sale_stock.py

```diff
@@ -307,6 +340,16 @@ def _compute_sale_id(self):
 
     sale_id = fields.Many2one(comodel_name='sale.order', string="Sale Order", compute='_compute_sale_id')
 
+    @api.multi
+    def _create_backorder(self, backorder_moves=[]):
+        res = super(StockPicking, self)._create_backorder(backorder_moves)
+        for picking in self.filtered(lambda pick: pick.picking_type_id.code == 'outgoing'):
```

Comment:
purpose is we need to set link of sale order in backorder of picking.but when we do partial picking and backorder is created we cannot identity any reference from so. so we need to search first all picking for outgoing, 
then find backorder of that picking and make order reference on back order. 

that why we need to search by outgoing picking here.